### ISSUE TITLE
Printing variable affects result

### ISSUE THREAD

**Issue by flexoron** [infradig/trealla#414]
```
?- L = "", X = L, Y = X.
   L = [], X = L, Y = L.

?- L = "", write(X = L), Y = X.
_1=[]   L = [], X = X.  % and Y?
```

**Comment by infradig** [comment]
Just a write(X) will do the same. Don't know why printing a variable
affects it's result display like this.

**Comment by flexoron** [comment]
right
```
?- write(X), Y = X.
_0  X = X.

?- write(Y), Y = X.
_0  X = Y.
```

**Comment by infradig** [comment]
Ok got it, writing X first (before Y is used) reorders the variables in the
environment so X is dumped first with the value Y, which is actually X (and
it print's as such). Something to look at.

**Comment by flexoron** [comment]
fixed with v1.20.10
